browser-sign-in/OktaBrowserSignInUITests/OktaBrowserSignInUITests.swift

```diff
@@ -15,93 +15,58 @@
  */
 
 import XCTest
+import WebAuthenticationUI
 
 class OktaBrowserSignInUITests: XCTestCase {
+    lazy var username: String? = {
+        ProcessInfo.processInfo.environment["LOGIN"]
+    }()
 
-    override func setUp() {
-        super.setUp()
-        clearKeyChain()
-        continueAfterFailure = false
+    lazy var password: String? = {
+        ProcessInfo.processInfo.environment["PASS"]
+    }()
+    
+    lazy var signInScreen: SignInScreen = { SignInScreen(self) }()
+    lazy var profileScreen: ProfileScreen = { ProfileScreen(self) }()
+
+    override func setUpWithError() throws {
         let app = XCUIApplication()
-        app.launchEnvironment = ProcessInfo.processInfo.environment
+        app.launchEnvironment = ["AutoCorrection": "Disabled"]
+        app.launchArguments = ["--reset-keychain"]
         app.launch()
+        
+        continueAfterFailure = false
     }
-
-    override func tearDown() {
-        super.tearDown()
+    
+    func testCancel() throws {
+        signInScreen.isVisible()
+        signInScreen.setEphemeral(true)
+        signInScreen.login()
+        signInScreen.cancel()
+        signInScreen.isVisible()
     }
 
-    func testLoginSuccess() {
-        let app = XCUIApplication()
-        let env = ProcessInfo.processInfo.environment
-        guard let login = env["LOGIN"], let pass = env["PASS"], let firstName = env["FIRST_NAME"] else {
-            XCTFail("Environment variables LOGIN, PASS, FIRST_NAME not set")
-            return
-        }
-        
-        app.buttons["Sign In"].tap()
-        passSystemAlert(button: "Continue")
-        
-        loginWith(login: login, and: pass)
-
-        XCTAssertTrue(app.staticTexts["Welcome, \(firstName)"].waitForExistence(timeout: 30))
-        
-        app.buttons["Sign Out"].tap()
-        
-        passSystemAlert(button: "Continue")
-        
-        XCTAssertTrue(XCUIApplication().staticTexts["Have an account?"].waitForExistence(timeout: 30))
+    func testEphemeralLoginAndSignOut() throws {
+        signInScreen.isVisible()
+        signInScreen.setEphemeral(true)
+        signInScreen.login(username: username, password: password)
+        profileScreen.wait()
+        let userNameQuery = profileScreen.app.staticTexts.matching(identifier: "Username").element.label
```

Comment:
I'd suggest moving this into the Page Object for the profile screen, to encapsulate the path to this element.